src/equisolve/utils/convert.py

```diff
@@ -0,0 +1,131 @@
+"""Functions for converting values into class:``equistore.TensorMap``s."""
+
+from typing import List
+
+import ase
+import numpy as np
+from equistore import Labels, TensorMap
+from equistore.block import TensorBlock
+
+
+def ase_to_tensormap(
+    frames: List[ase.Atoms], energy: str = None, forces: str = None, stress: str = None
+):
+    """Store informations from :class:``ase.Atoms`` in a class:``equistore.TensorMap``.
+
+    :param frames: ase.Atoms or list of ase.Atoms
+    :param energy: key for extracting energy per structure
+    :param forces: key for extracting atomic forces
+    :param stress: key for extracting stress per structure
+    """
+    if not isinstance(frames, list):
+        frames = [frames]
+
+    values = [f.info[energy] for f in frames]
+
+    if forces is not None:
+        positions_gradients = [-f.arrays[forces] for f in frames]
+    else:
+        positions_gradients = None
+
+    if stress is not None:
+        cell_gradients = [-f.arrays[stress] for f in frames]
+    else:
+        cell_gradients = None
+
+    return properties_to_tensormap(values, positions_gradients, cell_gradients)
+
+
+def properties_to_tensormap(
+    values: List[float],
+    positions_gradients: List[np.ndarray] = None,
+    cell_gradients: List[np.ndarray] = None,
+    is_structure_property: bool = True,
+):
+    """Create a class:``equistore.TensorMap`` from array like properties.
+
+    :values: List of length N
+    :positions_gradients: list of length N
+    :param cell_gradients: array of shape (N, 3, 3)
+    """
+
+    if not (is_structure_property):
+        raise NotImplementedError(
+            "Support for environment properties has not been implemented yet."
+        )
+
+    n_properties = len(values)
+
+    samples = Labels(["structure"], np.arange(n_properties).reshape(-1, 1))
+
+    block = TensorBlock(
+        values=np.asarray(values).reshape(-1, 1),
+        samples=samples,
+        components=[],
+        properties=Labels(["property"], np.array([(0,)])),
+    )
+
+    if positions_gradients is not None:
+        if n_properties != len(positions_gradients):
+            raise ValueError(
+                f"given {n_properties} values but "
+                f"{len(positions_gradients)} positions_gradients values"
+            )
+
+        data = np.concatenate(positions_gradients, axis=0)
```

Comment:
Can we add better name for this, we could just keep `positions_gradients` and overwrite and use later `n_properties` when we use the len, since we do a check that they are equal to each other

Reply:
No actually we can not use n_properties later. We need the number of atoms per structure. If we overwrite `positions_gradients` with `np.concatenate(positions_gradients, axis=0)` we loose this information.

Reply:
We could call it `gradient_data`?